closed/adds/jdk/src/windows/native/jdk/crypto/jniprovider/NativeCrypto_md.c

```diff
@@ -23,27 +23,48 @@
  */
 
 #include <windows.h>
+#include <stdio.h>
+#include <jni.h>
+#include <sys/stat.h>
 
 #include "NativeCrypto_md.h"
 
+#define JAVA_DLL "java.dll"
+
+/* Prototypes. */
+jboolean GetJREPath(char *path, jint pathsize);
+jboolean GetApplicationHome(char *buf, jint bufsize);
+
 /* Load the crypto library (return NULL on error) */
 void * load_crypto_library() {
     void * result = NULL;
     const char *libname;
     const char *oldname = "libeay32.dll";
+    char opensslpath[MAX_PATH];
+    char libpathname[MAX_PATH];
+
 
 #if defined (_WIN64)
     libname = "libcrypto-1_1-x64.dll";
 #else
     libname = "libcrypto-1_1.dll";
 #endif
 
-    result = LoadLibrary(libname);
-    
-    if (result == NULL) {
-        result = LoadLibrary(oldname);
-    }
+    if (GetJREPath(opensslpath, MAX_PATH)) {
+        sprintf(libpathname, "%s\\bin\\%s", opensslpath, libname);
```

Comment:
This may overflow the buffer: use `JLI_Snprintf `.

Reply:
Updated to use JLI_Snprintf and to check return code